### PR TITLE
Convert vaadin instrumentation to instrumenter api

### PR DESCRIPTION
**Pull request by laurit**
Part of #2713

### CODE REVIEW

File: instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/RpcInvocationHandlerInstrumentation.java [hunk 1]
```diff
@@ -53,21 +54,32 @@ public static void onEnter(
         @Advice.This RpcInvocationHandler rpcInvocationHandler,
         @Advice.Origin Method method,
         @Advice.Argument(1) JsonObject jsonObject,
+        @Advice.Local("otelRequest") VaadinRpcRequest request,
         @Advice.Local("otelContext") Context context,
         @Advice.Local("otelScope") Scope scope) {
 
-      context = tracer().startRpcInvocationHandlerSpan(rpcInvocationHandler, method, jsonObject);
+      Context parenContext = Java8BytecodeBridge.currentContext();
```
Comment (mateuszrzeszutek) [inline instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/RpcInvocationHandlerInstrumentation.java:61]: Typo: `parenContext` -> `parentContext`

File: instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinHelper.java [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+    Context parentContext = Context.current();
+    if (!serviceInstrumenter.shouldStart(parentContext, request)) {
+      return null;
+    }
+
+    return serviceInstrumenter.start(parentContext, request);
+  }
+
+  public void endVaadinServiceSpan(
+      Context parentContext, VaadinServiceRequest request, Throwable throwable) {
```
Comment (mateuszrzeszutek) [inline instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinHelper.java:41]: Shouldn't it be just `context`, not `parentContext`? Same in `endRequestHandlerSpan`

File: instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinHelper.java [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+    serviceInstrumenter.end(parentContext, request, null, throwable);
+
+    VaadinServiceContext vaadinServiceContext = parentContext.get(SERVICE_CONTEXT_KEY);
```
Comment (mateuszrzeszutek) [inline instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinHelper.java:44]: Can this call be moved to `getSpanNameForVaadinServiceContext()`?

**Review by mateuszrzeszutek — APPROVED** [review]

File: instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinSingletons.java [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+  private static final Instrumenter<VaadinServiceRequest, Void> SERVICE_INSTRUMENTER;
+  private static final VaadinHelper HELPER;
+
+  static {
+    CLIENT_CALLABLE_INSTRUMENTER =
+        Instrumenter.<VaadinClientCallableRequest, Void>builder(
+                GlobalOpenTelemetry.get(),
+                INSTRUMENTATION_NAME,
+                VaadinSingletons::clientCallableSpanName)
+            .setDisabled(ExperimentalConfig.get().suppressControllerSpans())
```
Comment (trask) [inline instrumentation/vaadin-14.2/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/vaadin/VaadinSingletons.java:35]: 👍

**Review by trask — APPROVED** [review]
🥳

**Pull request merged**